tests/unit/helpers/class-wc-helper-intention.php

```diff
@@ -34,25 +34,60 @@ public static function create_charge( $data = [] ) {
 						'funding' => 'credit',
 					],
 				],
+				'payment_method'         => 'pm_mock',
+				'amount_captured'        => 5000,
+				'amount_refunded'        => 0,
+				'application_fee_amount' => 0,
+				'balance_transaction'    => 'txn_mock',
```

Comment:
@dmvrtx Stripe typically supplies a [balance_transaction](https://stripe.com/docs/api/balance_transactions/object) object attached to this property rather than directly its ID. I'm curious if you set it to txn_xxx id here on purpose.

**Context:** In [trying to access](https://github.com/Automattic/woocommerce-payments/blob/fbc8664dfb36ab954c5961dea6941868716ffe18/includes/class-wc-payments-order-service.php#L791) the balance transaction id, my PR is [failing](https://github.com/Automattic/woocommerce-payments/actions/runs/7338045342/job/19979943150?pr=7945) 25 tests 😕 However, manual testing confirms that my code works with actual payments.

Reply:
I just pushed this change in my PR. Hope you're okay with it.

https://github.com/Automattic/woocommerce-payments/pull/7945/commits/83d5af89335bd77c2cdcb9fa58d3418c21d998e7

Reply:
Stripe documentation indicates that it is a transaction ID e.g. [for a Charge object](https://stripe.com/docs/api/charges/object#charge_object-balance_transaction)

I think failing tests are based on assumption that request to Stripe is sent with an [`expand` setting](https://stripe.com/docs/api/expanding_objects) for the `balance_transaction`. In this case it is indeed returned as an object. Such nuance is not handled well by the testing framework though.

Reply:
TIL about the `expand` setting.